lib/dao.js

```diff
@@ -3089,7 +3089,7 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
 
         function replaceCallback(err, data) {
           if (err) return cb(err);
-          context.data = data;
+          if (data) context.data = data;
           if (typeof connector.generateContextData === 'function') {
             context = connector.generateContextData(context, data);
```

Comment:
Is it possible to change this to be callback? Because `generateContextData ` might need query db to get result?

Reply:
Let's leave such change out of scope of this pull request please. We need to fix the regression introduced by #1598 as soon as possible.

Changing `generateContextData` into an async function would be more involved, because we need to preserve backwards compatibility with existing connectors that provide a sync version only.